### PR TITLE
Feature/ADF-488/Add equal method to dependd on property collections

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-488](https://oat-sa.atlassian.net/browse/ADF-488)_

### CODE REVIEW

File: core/resource/DependsOnPropertyCollection.php [hunk 1]
```diff
@@ -26,8 +26,27 @@
 use core_kernel_classes_Property;
 
 /**
- * @method core_kernel_classes_Property current()
+ * @method core_kernel_classes_Property|null current()
  */
 class DependsOnPropertyCollection extends ArrayIterator
 {
+    public function isEqual(DependsOnPropertyCollection $dependsOnPropertyCollection): bool
```
Comment (gabrielfs7) [inline core/resource/DependsOnPropertyCollection.php:33]: Maybe now it makes sense some test?

**Review by gabrielfs7 — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: core/resource/DependsOnPropertyCollection.php (hunk 1 above)
Comment (shpran) [reply core/resource/DependsOnPropertyCollection.php:33]: Added 🙂

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**